src/trace/trace.cpp

```diff
@@ -159,7 +162,16 @@ void Trace::begin_record()
 
 void Trace::end_record()
 {
-    stopping_time_ = time::now();
+    // write missing program_end events
+    auto now = lo2s::time::now();
```

Comment:
doesn't that technically require the same trick lick here: https://github.com/tud-zih-energy/lo2s/blob/bcf7f78f7c10be11c2c89da31de2075df02fa98e/src/perf/sample/writer.cpp#L212